From d8f7382ce1847e30aecf10b10fa21b4273d53323 Mon Sep 17 00:00:00 2001
From: Mikolaj <mikolaj.konarski@gmail.com>
Date: Tue, 14 May 2013 17:03:43 +0200
Subject: [PATCH 6/9] Add last commit short hash to ghc version name

E.g., 7.4.2.dae48ce. You need to do perl boot; ./configure the first
time you compile ghc and ./configure each subsequent time to get the commit
hashes in your version string. The trick mostly invented by edsko.
---
 aclocal.m4 | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index b572e32..4de9954 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1460,18 +1460,13 @@ AC_SUBST([GccExtraViaCOpts],$fp_cv_gcc_extra_opts)
 # ---------------------
 AC_DEFUN([FP_SETUP_PROJECT_VERSION],
 [
-if test "$RELEASE" = "NO"; then
+if test "$RELEASE" = "YES"; then
     AC_MSG_CHECKING([for GHC version date])
     if test -f VERSION_DATE; then
         PACKAGE_VERSION=${PACKAGE_VERSION}.`cat VERSION_DATE`
         AC_MSG_RESULT(given $PACKAGE_VERSION)
     elif test -d .git; then
-        changequote(, )dnl
-        ver_date=`git log -n 1 --date=short --pretty=format:%ci | cut -d ' ' -f 1 | tr -d -`
-        if echo $ver_date | grep '^[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]$' 2>&1 >/dev/null; then true; else
-        changequote([, ])dnl
-                AC_MSG_ERROR([failed to detect version date: check that git is in your path])
-        fi
+        ver_date=`git rev-parse --short HEAD`
         PACKAGE_VERSION=${PACKAGE_VERSION}.$ver_date
         AC_MSG_RESULT(inferred $PACKAGE_VERSION)
     elif test -d _darcs; then
-- 
1.7.12.4 (Apple Git-37)

